Dockerfile

```diff
@@ -1,14 +1,79 @@
-FROM alpine:3.6
+########################################################################################
+# Inherit from this docker image and copy your project's commands into the
+# /code/commands directory. Any other supporting files can go in the /code
+# directory. The /code directory will be added to PYTHONPATH.
+########################################################################################
 
-RUN apk add --update python py-pip vim curl bash 
 
-WORKDIR /code
+### BUILD POETRY PACKAGE MANAGER AND INSTALL DEPENDENCIES ##############################
+# This downloads, installs and configures the poetry tool under the root user in the
+# build image. It then installs the python package dependencies specified in the
+# pyproject.toml file according to the frozen versions in poetry.lock.
+########################################################################################
+FROM python:3.8-slim as builder
+
+# Install packages required to build native library components
+RUN apt-get update \
+ && apt-get -y --no-install-recommends install \
+    gettext \
+    gcc \
+    g++ \
+    make \
+    libc6-dev \
+ && rm -rf /var/lib/apt/lists/*
+
+# Upgrade pip
+RUN pip install --upgrade pip
+
+# Configure pip
+COPY pip.conf /etc/pip.conf
+
+# Install poetry
+ARG POETRY_VERSION=1.0.5
+ADD https://raw.githubusercontent.com/python-poetry/poetry/${POETRY_VERSION}/get-poetry.py /tmp/get-poetry.py
+RUN python /tmp/get-poetry.py
```

Comment:
why get-poetry.py, and not just pip install poetry, given that we've just upgraded pip?